### PR TITLE
new  qc json for trd

### PR DESCRIPTION
**Pull request by bazinski**

### CODE REVIEW

File: MC/config/QC/json/trd-digits-task.json [hunk 1]
```diff
@@ -9,70 +9,139 @@
...
+          "query": "digits:TRD/DIGITS;tracklets:TRD/TRACKLETS;triggers:TRD/TRKTRGRD;rawstats:TRD/RAWSTATS"
+        },
+        "taskParameters": {
+          "peakregionstart": "7.0",
+          "peakregionend": "20.0",
+          "pulseheightpeaklower": "1.0",
+          "pulseheightpeakupper": "5.0"
+        },
+        "location": "remote",
+	"saveObjectsToFile":"qcrootobjects.root"
```
Comment (knopers8) [inline MC/config/QC/json/trd-digits-task.json:52]: Please disable saving objects to a file (remove this line), so QC on grid does not generate more files than necessary. The same comment applies to other tasks.

File: MC/config/QC/json/trd-digits-task.json [hunk 2]
```diff
@@ -9,70 +9,139 @@
...
+        "active": "false",
+        "className": "o2::quality_control_modules::trd::RawData",
+        "moduleName": "QcTRD",
+        "detectorName": "TRD",
+        "cycleDurationSeconds": "60",
+        "maxNumberCycles": "-1",
+        "dataSource_comment": "no comment",
+        "dataSource": {
+          "type": "direct",
+          "query": "digits:TRD/DIGITS;tracklets:TRD/TRACKLETS;triggers:TRD/TRKTRGRD;rawstats:TRD/RAWSTATS"
```
Comment (knopers8) [inline MC/config/QC/json/trd-digits-task.json:43]: Are you sure that `rawstats:TRD/RAWSTATS` is accesible at this stage of computation (or in simulated data at all)?
Here is what TRD QC requests to be present:
https://github.com/AliceO2Group/O2DPG/blob/ca1971d3835f9f5186f488993281e3960fecdd95/MC/bin/o2dpg_sim_workflow.py#L785
Here is the documentation about adding QC to MC scripts:
https://github.com/AliceO2Group/O2DPG/tree/master/MC#adding-qc-tasks-to-the-simulation-script

File: MC/config/QC/json/trd-digits-task.json [hunk 3]
```diff
@@ -9,70 +9,139 @@
...
     "tasks": {
-      "QcTask": {
+      "RawDataTask": {
+        "active": "false",
+        "className": "o2::quality_control_modules::trd::RawData",
+        "moduleName": "QcTRD",
+        "detectorName": "TRD",
+        "cycleDurationSeconds": "60",
+        "maxNumberCycles": "-1",
+        "dataSource_comment": "no comment",
```
Comment (knopers8) [inline MC/config/QC/json/trd-digits-task.json:40]: you can delete the line in such case.

File: MC/config/QC/json/trd-digits-task.json [hunk 4]
```diff
@@ -9,70 +9,139 @@
...
+          "type": "direct",
+          "query": "digits:TRD/DIGITS;tracklets:TRD/TRACKLETS;triggers:TRD/TRKTRGRD;rawstats:TRD/RAWSTATS"
+        },
+        "taskParameters": {
+          "peakregionstart": "7.0",
+          "peakregionend": "20.0",
+          "pulseheightpeaklower": "1.0",
+          "pulseheightpeakupper": "5.0"
+        },
+        "location": "remote",
```
Comment (knopers8) [inline MC/config/QC/json/trd-digits-task.json:51]: this is not applicable in the case of batch processing, you can delete all "location" lines.

**Review by knopers8 — CHANGES_REQUESTED** [review]
Thanks for the update. I see that you added some new tasks, which in principle should be fine, since the tasks request the same input to be present at this stage. This is not entirely true for the `RawDataTask`, see the corresponding comment.
Additionally, please consider applying also the other suggestions.

**Comment by bazinski** [comment]
Hi Piotr, ja its clear I just took standard qc file and replaced the old non functioning one and switched off the rawdata task which receives a stats message directly from the raw data reader, and hence not relevant to MC.  
Will fix now, and disable pulsheight task which i think is irrelevant for MC.

**Comment by knopers8** [comment]
Hi @bazinski, could you please apply the changes? Unless you are still on holidays, then sorry for bothering you.

**Comment by bazinski** [comment]
HI @knopers8 oops this completely fell off the todo list. 
I removed the tracklet qc after I looked at the python and did not want to touch, maybe later.
All your comments done.

File: MC/config/QC/json/trd-digits-task.json (hunk 3 above)
Comment (bazinski) [reply MC/config/QC/json/trd-digits-task.json]: done

File: MC/config/QC/json/trd-digits-task.json (hunk 2 above)
Comment (bazinski) [reply MC/config/QC/json/trd-digits-task.json]: rawstats would not work no, and also the tracklet task is removed for not wanting to edit the python.

File: MC/config/QC/json/trd-digits-task.json (hunk 1 above)
Comment (bazinski) [reply MC/config/QC/json/trd-digits-task.json]: done

File: MC/config/QC/json/trd-digits-task.json [hunk 5]
```diff
@@ -9,70 +9,139 @@
...
+        "passName": "",             "": "Pass type - e.g. spass, cpass1",
+        "provenance": "qc",         "": "Provenance - qc or qc_mc depending whether it is normal data or monte carlo data"
       },
       "monitoring": {
         "url": "infologger:///debug?qc"
       },
       "consul": {
-        "url": "http://localhost:8500"
+        "url": "http://consul-test.cern.ch:8500"
       },
```
[the remark this answers is not in the document]
Comment (bazinski) [reply MC/config/QC/json/trd-digits-task.json]: done

**Review by knopers8 — APPROVED** [review]
Thank you!

**Pull request merged**